### PR TITLE
Declarative Refactor

### PR DESCRIPTION
**Pull request by Bernstern**
Started https://www.notion.so/Declarative-Refactor-584cad39024f47b1ab594f62d5f9c2df?pvs=4

Still work needed but should be used as starting point for future improvement

### CODE REVIEW

File: get_started.ps1 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Setup our quail submodule
+Write-Host "Setting up quail submodule" -ForegroundColor Blue
+git submodule update --init --recursive
+Write-Host "Quail submodule setup complete" -ForegroundColor Green
+
+# Symlink the quail submodule
+Write-Host "Symlinking quail submodule" -ForegroundColor Blue
```
Comment (Bernstern) [inline get_started.ps1:7]: Get rid of these

File: simgui-ds.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
+{
```
Comment (Bernstern) [inline simgui-ds.json]: Delete file

File: networktables.json [hunk 3]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (Bernstern) [inline networktables.json]: Delete file

File: simgui.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
+{
```
Comment (Bernstern) [inline simgui.json]: Delete file

File: src/main/java/frc/robot/Robot.java [hunk 5]
```diff
@@ -77,11 +131,34 @@ public void teleopInit() {
     if (m_autonomousCommand != null) {
       m_autonomousCommand.cancel();
     }
+
```
Comment (Bernstern) [inline src/main/java/frc/robot/Robot.java:117]: Remove

File: src/main/java/frc/robot/Robot.java [hunk 6]
```diff
@@ -77,11 +131,34 @@ public void teleopInit() {
...
+		// driveCommand dt = new driveCommand(primaryController, gyro, drivetrain);
+		// CommandScheduler.getInstance().schedule(dt);
   }
 
   /** This function is called periodically during operator control. */
   @Override
-  public void teleopPeriodic() {}
+  public void teleopPeriodic() {
+    // double velo = controller.getLeftTriggerAxis();
```
Comment (Bernstern) [inline src/main/java/frc/robot/Robot.java:125]: Remove

File: src/main/java/frc/robot/RobotContainer.java [hunk 7]
```diff
@@ -4,51 +4,99 @@
...
+ * the {@link Robot}
+ * periodic methods (other than the scheduler calls). Instead, the structure of
+ * the robot (including
  * subsystems, commands, and trigger mappings) should be declared here.
  */
 public class RobotContainer {
-  // The robot's subsystems and commands are defined here...
-  private final ExampleSubsystem m_exampleSubsystem = new ExampleSubsystem();
+  // Initialize the gyro module
+  private final GyroModule m_gyro = new GyroModule();
```
Comment (Bernstern) [inline src/main/java/frc/robot/RobotContainer.java]: @Bernstern is poking around w/ other ways to place the gyro that is friendly to use

**Review by Bernstern — COMMENTED** [review]
Not done yet

**Comment by Mineinjava** [comment]
Does the quail submoduling work on mac? 

I think installing quail as a submodule may be a good thing to put in the quail documentation.

**Comment by Bernstern** [comment]
> Does the quail submoduling work on mac?
> 
> I think installing quail as a submodule may be a good thing to put in the quail documentation.

The relative path should work for most shells, might have a line ending issue though since windows is crlf

**Comment by Mineinjava** [comment]
Forgot to push today. After tomorrow we should be good to merge.

**Pull request merged**